src/renderer/helpers/copyAndPaste.ts

```diff
@@ -98,9 +102,37 @@ export const pasteFromClipboard = (
                 case 'text/rtf':
                     log.debug('Clipboard rtf', clipboard.readRTF());
                     break;
-                case 'image/png':
-                    log.debug('Clipboard image', clipboard.readImage().toPNG());
+                case 'image/jpeg':
+                case 'image/gif':
+                case 'image/bmp':
+                case 'image/tiff':
+                case 'image/png': {
+                    const tempDir = os.tmpdir();
+                    const imgData = clipboard.readImage().toPNG();
+                    const randName =
+                        (Math.random() + 1).toString(36).substring(7) +
+                        (Math.random() + 1).toString(36).substring(7) +
+                        (Math.random() + 1).toString(36).substring(7);
+                    const imgPath = path.join(tempDir, `chaiNNer-clipboard-${randName}.png`);
+                    writeFile(imgPath, imgData)
+                        .then(() => {
+                            log.debug('Clipboard image', imgPath);
+                            createNode({
+                                nodeType: 'regularNode',
+                                position: { x: 0, y: 0 },
```

Comment:
Maybe we could place the node at the center of the current viewport instead of the origin? Right now, users might not even see the new node.

Reply:
That's a good point